### PR TITLE
sql: add job id to ROLL BACK job for schema change

### PR DESCRIPTION
**Pull request by vivekmenezes**
this will allow a user to know what original job
the roll back job is referring to.

Release note: None

### CODE REVIEW

**Comment by vivekmenezes** [comment]
I'll fix the tests once we agree what this should look like

File: pkg/sql/schema_changer.go [hunk 1]
```diff
@@ -1026,7 +1026,7 @@ func (sc *SchemaChanger) createRollbackJob(
 			}
 			payload := job.Payload()
 			rollbackJob := sc.jobRegistry.NewJob(jobs.Record{
-				Description:   "ROLL BACK " + payload.Description,
+				Description:   fmt.Sprintf("ROLL BACK JOB: %d, ", *job.ID()) + payload.Description,
```
Comment (mjibson) [inline pkg/sql/schema_changer.go]: This makes the output something like `ROLL BACK JOB: 123 ALTER TABLE BLAH`. Maybe `ROLL BACK JOB 123: ALTER TABLE BLAH` is more readable?

Comment (mjibson) [reply pkg/sql/schema_changer.go]: But I don't care too much either way.

**Review by vivekmenezes — COMMENTED** [review]
Added all the tests to the change. TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28857#-:-LKO8bYlEjg6MLydD3mL:b-o7s2k3)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/schema_changer.go, line 1029 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28857#-LKNqLTznSDQNVuwqEdy-r1-1029:-LKO8_b6-vYOuCga1XiN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e9d7e0ae99b7ebc13df378f723e439ea3dcf71ba/pkg/sql/schema_changer.go#L1029)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

But I don't care too much either way.
</blockquote></details>

Done.

File: pkg/sql/schema_changer.go [hunk 2]
```diff
@@ -1026,7 +1026,7 @@ func (sc *SchemaChanger) createRollbackJob(
 			}
 			payload := job.Payload()
 			rollbackJob := sc.jobRegistry.NewJob(jobs.Record{
-				Description:   "ROLL BACK " + payload.Description,
+				Description:   fmt.Sprintf("ROLL BACK JOB %d: ", *job.ID()) + payload.Description,
```
Comment (mjibson) [inline pkg/sql/schema_changer.go]: would using %s here make more sense than the `+`? same below.

**Review by madelynnblue — COMMENTED** [review]
workload-linux seems to be an unintended included file

**Review by vivekmenezes — COMMENTED** [review]
taken workload-linux out!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28857#-:-LKRsAZm0q3JWMI-ayX4:b-qz01rd)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/schema_changer.go, line 1029 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/28857#-LKNqLTznSDQNVuwqEdy-r2-1029:-LKRsDQw4OWRXbwn6A3e:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/405232915c9e4259572f9dd4e5aa46d51630f1bc/pkg/sql/schema_changer.go#L1029)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

would using %s here make more sense than the `+`? same below.
</blockquote></details>

Done.

File: pkg/sql/schema_changer.go (hunk 2 above)
Comment (mjibson) [reply pkg/sql/schema_changer.go]: Missed this one.

**Comment by vivekmenezes** [comment]
@mjibson this look okay? Thanks

**Review by madelynnblue — APPROVED** [review]
Sorry. This slipped off my email.

**Pull request merged**